integrations/clients/hud_income_limits/client.py

```diff
@@ -0,0 +1,287 @@
+        """Fetch data from cache or API and cache the result."""
+        data = cache.get(cache_key)
+
+        if not data:
+            params = {"year": str(year)} if year else {}
+            data = self._api_request(endpoint, params)
+            cache.set(cache_key, data, self.CACHE_TTL)
+
+        return data
+
+    def _validate_data_response(self, data: dict, county: str, state_code: str) -> dict:
+        """Validate API response contains data and return the data section."""
+        if not data or "data" not in data:
+            raise HudIncomeClientError(f"No income limit data found for {county}, {state_code}")
+        return data["data"]
+
+    def _get_entity_id(self, state_code: str, county_name: str, year: int) -> str:
+        """Get FIPS entity ID for a county in any state."""
+        # Normalize county name
+        county_name = county_name.strip()
+        if not county_name.lower().endswith("county"):
+            county_name = f"{county_name} County"
+
+        # Check cache
+        cache_key = f"hud_counties_{state_code}_{year or 'latest'}"
+        counties = cache.get(cache_key)
+
+        if not counties:
+            # Use FMR endpoint to list counties (shared across FMR and IL APIs)
+            # Note: This requires FMR dataset API access in addition to IL dataset
+            # Per HUD API docs: https://www.huduser.gov/portal/dataset/fmr-api.html
+            # - 'year' parameter: optional, retrieves data for specific year
+            # - 'updated' parameter: for 2025+, gets refreshed FIPS codes
+            params = {}
+            if year:
+                params["year"] = str(year)
+                if year >= 2025:
+                    params["updated"] = str(year)
+            counties = self._api_request(f"fmr/listCounties/{state_code.upper()}", params)
+            cache.set(cache_key, counties, self.CACHE_TTL)
+
+        # Find matching county (FMR API returns array directly, not wrapped in data object)
+        if not counties or not isinstance(counties, list):
+            raise HudIncomeClientError(f"Could not retrieve counties for {state_code}")
+
+        for county in counties:
+            # HUD API changed field name in 2025 dataset update:
+            # - Without 'updated' param or pre-2025: uses 'county_name'
+            # - With 'updated=2025' param: uses 'cntyname' (shortened field name)
+            # We check both for compatibility across all years
+            name = county.get("county_name") or county.get("cntyname", "")
+            if name.lower() == county_name.lower():
+                return county["fips_code"]
+
+        raise HudIncomeClientError(f"County not found: {county_name}, {state_code}")
+
+    def _api_request(self, endpoint: str, params: Optional[dict] = None) -> dict:
+        """Make an API request to HUD."""
+        try:
+            response = requests.get(f"{self.BASE_URL}/{endpoint}", headers=self.headers, params=params, timeout=30)
```

Comment:
Think this should be fine for now but might be worth keeping an eye on whether we experience transient errors. If we do, maybe we consider retry logic with exponential backoff.